### PR TITLE
Switch config-driven autoclustering to rabbit_peer_discovery_backend

### LINKED ISSUE #486: Pluggable cluster member discovery
Currently we support two ways to discover cluster members:

 * None (manual clustering)
 * `rabbit.cluster_nodes` ("autoclustering")

The excellent [autocluster](https://github.com/aweber/rabbitmq-autocluster/) plugin provides an API and several more implementations (DNS A record-based discovery, Consul, etcd). We need to integrate this behaviour into the core and refactor our own "autoclustering" to use it. A lot of users would be interested in EC2 API-based discovery, à la ElasticSearch AWS Cloud plugin.

Note that this issue is only about discovering nodes. How clusters are formed is a separate concern, and autocluster is different from rabbitmq-clusterer, for example. We need to find a way to make both possible but that belongs to a separate issue: #487.

### PR DESCRIPTION
**Pull request by michaelklishin**
This switches the classic automatic clustering feature for virgin nodes to the newly introduced `rabbit_peer_discovery_backend` behaviour pulled in from https://github.com/aweber/rabbitmq-autocluster.

Fixes #486. Depends on #987. References #487.

### CODE REVIEW

**Comment by michaelklishin** [comment]
This depends on #996 (or rather, makes a lot more sense to merge after it) but otherwise seems to be ready for evaluation.

**Comment by michaelklishin** [comment]
@gmr FYI.

File: src/rabbit_peer_discovery.erl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+%%
+%% Software distributed under the License is distributed on an "AS IS"
+%% basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See
+%% the License for the specific language governing rights and
+%% limitations under the License.
+%%
+%% The Original Code is RabbitMQ.
+%%
+%% The Initial Developer of the Original Code is Pivotal Software, Inc.
+%% Copyright (c) 2007-2015 Pivotal Software, Inc.  All rights reserved.
```
Comment (kjnilsson) [inline src/rabbit_peer_discovery.erl]: 2016?

**Comment by michaelklishin** [comment]
Ready for another round.

**Comment by acogoluegnes** [comment]
I tested by listing cluster nodes in the configuration file (the old way) and it works as before. LGTM.

**Pull request merged**